app/org/sagebionetworks/bridge/dynamodb/DynamoUploadSchemaDao.java

```diff
@@ -101,7 +103,24 @@ public void setStudyIdIndex(DynamoIndexHelper studyIdIndex) {
         // Get the current rev.
         String studyId = studyIdentifier.getIdentifier();
         String schemaId = survey.getIdentifier();
-        int oldRev = getCurrentSchemaRevision(studyId, schemaId);
+        DynamoUploadSchema oldUploadSchema = getUploadSchemaNoThrow(studyId, schemaId);
+        int oldRev;
+        if (oldUploadSchema != null) {
+            oldRev = oldUploadSchema.getRevision();
+
+            // Optimization: If the new schema and the old schema have the same fields, return the old schema instead
+            // of creating a new one.
+            //
+            // Dump the fieldDefLists into a set, because if we have the same fields in a different order, the schemas
+            // are compatible, and we should use the old schema too.
+            Set<UploadFieldDefinition> oldFieldDefSet = ImmutableSet.copyOf(oldUploadSchema.getFieldDefinitions());
+            Set<UploadFieldDefinition> newFieldDefSet = ImmutableSet.copyOf(fieldDefList);
+            if (oldFieldDefSet.equals(newFieldDefSet)) {
+                return oldUploadSchema;
+            }
+        } else {
+            oldRev = 0;
```

Comment:
I think the optimizer will optimize this away either way, but I can see this from a readability perspective. Changed.